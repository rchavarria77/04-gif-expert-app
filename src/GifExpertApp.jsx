import { useState } from 'react';
import { AddCategory, GifGrid } from './Components';

const GifExpertApp = () => {
  const [categories, setCategories] = useState(['One Punch']);

  const onAddCategory = (newCategory) => {
    if (categories.find((category) => category.toLowerCase() === newCategory.toLowerCase())) return;

    setCategories([newCategory, ...categories]);
  };

  return (
    <>
      <h1>Gif Expert App</h1>
      <AddCategory onNewCategory={onAddCategory} />

      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};

export default GifExpertApp;
