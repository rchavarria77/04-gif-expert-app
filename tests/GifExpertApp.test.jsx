import { render, screen } from '@testing-library/react';
import GifExpertApp from '../src/GifExpertApp';

describe('Testing <GifExpertApp /> component', () => {
  test('debe de hacer match con el snapshot ', () => {
    const { container } = render(<GifExpertApp />);

    expect(container).toMatchSnapshot();
  });

  test('debe de mostrar el titulo en el componente', () => {
    render(<GifExpertApp />);

    expect(screen.getByText('Gif Expert App')).toBeTruthy();
  });
});
