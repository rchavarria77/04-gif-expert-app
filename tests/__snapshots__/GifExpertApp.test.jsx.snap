// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifExpertApp /> component debe de hacer match con el snapshot  1`] = `
<div>
  <h1>
    Gif Expert App
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    One Punch
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
